addon/models/node.js

```diff
@@ -105,5 +70,25 @@ export default OsfModel.extend({
     root: DS.belongsTo('node', {
         inverse: null
     }),
-    logs: DS.hasMany('logs')
+    logs: DS.hasMany('logs'),
+
+    save() {
+        // Some duplicate logic from osf-model#save needed to support
+        // contributor edits being saved through the node
+        var promise = this._super(...arguments);
+        var contributors = this.hasMany('contributors').hasManyRelationship;
+        if (contributors.hasData || contributors.hasLoaded) {
+            this.set(
+                '_dirtyRelationships.contributors.update',
+                contributors.members.list.filter(m => !m.record.get('isNew') && Object.keys(m.record.changedAttributes()).length > 0)
+            );
+            // Contributors are a 'real' delete, not just a de-reference
+            this.set(
+                '_dirtyRelationships.contributors.delete',
+                this.get('_dirtyRelationships.contributors.remove')
+            );
+            this.set('_dirtyRelationships.contributors.remove', []);
+        }
+        return promise;
```

Comment:
> Is there a situation where one of those changes could fail, but not the other? 

This is at least partially taken into account. If you [look at the internals of _setupRelationships](https://github.com/emberjs/data/blob/master/addon/-private/system/store.js#L2455), it only updates the fields for which data is included. So the strategy here is to be conservative about what we tell the store has updated.

 One important note: if the inverse model is dirty, `OsfModel#save` _always_ returns the response from the super call. Looking more at the Store's internals:
See 
- https://github.com/emberjs/data/blob/master/addon/-private/system/store.js#L1646
- https://github.com/emberjs/data/blob/master/addon/-private/system/store.js#L2430
This should be a reliable way to communicate state changes. 

Admittedly this is imperfect and will require future improvements.